tests/sklearn/test_sklearn_models.py

```diff
@@ -1120,6 +1137,206 @@ def check_exposition_structural_methods_decision_trees(model, x, y):
     )
 
 
+# pylint: disable=too-many-locals,too-many-statements
+@pytest.mark.parametrize("model_class, parameters", get_sklearn_tree_models_and_datasets())
+@pytest.mark.parametrize("use_rounding", [False, True])
+def test_load_fitted_sklearn_tree_models(
+    subtests,
+    model_class,
+    parameters,
+    use_rounding,
+    load_data,
+    is_weekly_option,
+    verbose=True,
+):
+    """Test `from_sklearn_model` functionnality of tree-based models."""
+
+    numpy.random.seed(0)
+    os.environ["TREES_USE_ROUNDING"] = str(int(use_rounding))
+
+    x, y = get_dataset(
+        model_class, parameters, min(N_BITS_REGULAR_BUILDS), load_data, is_weekly_option
+    )
+
+    if verbose:
+        print("Run check_load_pre_trained_sklearn_models")
+
+    assert issubclass(model_class, BaseTreeEstimatorMixin)
+    concrete_model = instantiate_model_generic(model_class, n_bits=min(N_BITS_REGULAR_BUILDS))
+    # Fit the model and retrieve both the Concrete ML and the scikit-learn models
+    with warnings.catch_warnings():
+        # Sometimes, we miss convergence, which is not a problem for our test
+        warnings.simplefilter("ignore", category=ConvergenceWarning)
+        concrete_model, sklearn_model = concrete_model.fit_benchmark(x, y)
+
+    # This step is needed in order to handle partial classes
+    model_class = get_model_class(model_class)
+    max_n_bits = 18
+    reasonable_n_bits = 8
+
+    if is_model_class_in_a_list(
+        model_class,
+        _get_sklearn_tree_models(classifier=True, regressor=False),
+    ):
+        for n_bits, cml_tolerance, sklearn_tolerance in [
+            (max_n_bits, 1e-1, 1e-7),
+            (reasonable_n_bits, 6e-2, 6e-2),
+        ]:
+            # Load a Concrete ML model from the fitted scikit-learn one
+            loaded_from_threshold = model_class.from_sklearn_model(
+                sklearn_model,
+                X=None,
+                n_bits=n_bits,
+            )
+
+            loaded_from_data = model_class.from_sklearn_model(
+                sklearn_model,
+                X=x,
+                n_bits=n_bits,
+            )
+
+            # Compile both the initial Concrete ML model and the loaded one
+            concrete_model.compile(x)
+            mode = "disable"
```

Comment:
I think you should use simulate here

Reply:
Ho yeah forgot to activate simulation mode with reasonable number of bits